content/en/docs/components/training/reference/architecture.md

```diff
@@ -0,0 +1,33 @@
++++
+title = "Architecture"
+description = "The Training Operator Architecture"
+weight = 10
++++
+
+{{% stable-status %}}
+
+## What is the Training Operator Architecture?
+
+The original design was drafted in April 2021 and is [available here for reference](https://docs.google.com/document/d/1x1JPDQfDMIbnoQRftDH1IzGU0qvHGSU4W6Jl4rJLPhI/).
+The goal was to provide a unified Kubernetes operator that supports multiple
+machine learning/deep learning frameworks. This was done by having a "Frontend"
+operator that decomposed the job into different configurable Kubernetes
+components (e.g., Role, PodTemplate, Fault-Tolerance, etc.). The "Backend"
+operator watches all Role Customer Resources and manages pod performance.
+
+The benefits of this approach were:
+1. Shared testing and release infrastructure
+2. Unlocked production manifests and metadata support
+3. Simpler Kubeflow releases
+4. A Single Source of Truth (SSOT) for other Kubeflow components to interact with
+
+The V1 Training Operator architecture diagram can be seen in the diagram below:
+
+<img src="/docs/components/training/images/training-operator-v1-architecture.drawio.svg"
+  alt="Training Operator V1 Architecture"
+  class="mt-3 mb-3">
+
+The diagram displays PyTorch and its configured communication methods but it
+is worth mentioning that each framework can have its own appraoch(es) to
+communicating across pods. Additionally, each framework can have its own set of
```

Comment:
@kubeflow/wg-training-leads Don't we use the same communication method for all other frameworks ? E.g. creation of service for each Pod to setup the networking.

Reply:
As an example, PyTorch has several available [Communication Backends](https://pytorch.org/tutorials/intermediate/dist_tuto.html#communication-backends), which includes:

1. Gloo
2. MPI 
3. NCCL
4. UCC (Experimental)

Note the link above actually doesn't reflect UCC as that's in the actual source code and can be found in the `torch.distributed.init_process_group` [documentation](https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group).

But maybe I misunderstood the documentation, so let me know.

Reply:
@andreyvelich Yes, that's right.
@franciscojavierarceo The TrainingOperator sets up the Pod-to-Pod communications from the infrastructure perspective. The training backend communication techniques (Gloo, MPI, NCCL, etc,..) depend on the user's training codes.

In conclusion, @andreyvelich mentions about communications from infrastructure perspective like K8s, Training-Operator, but you mentioned about communications from training codes perspective.

The communications from an infrastructure perspective are used to execute the communications from the training codes perspective. So, the TraininOperator never set up the communications from the training code perspective.

Setting up communications from training code is responsible for end users like researchers.

Reply:
I'll add this nuance to the docs, I think it's useful clarification from a security perspective. Thanks @tenzen-y!

Reply:
Updated. Let me know what you all think.